packit/utils/changelog_helper.py

```diff
@@ -41,24 +41,35 @@ def entry_from_action(self) -> Optional[str]:
 
     def update_dist_git(self, full_version: str, upstream_tag: str) -> None:
         """
-        Updates changelog when running `update-dist-git`.
+        Update the spec-file in dist-git by setting the 'Version' tag and
+        adding a new entry in the %changelog section.
+        If downstream spec file has the %autochangelog macro then
+        preserve it and do not write a comment to the %changelog section.
 
         Args:
             full_version: Version to be set in the spec-file.
             upstream_tag: The commit messages after last tag and before this tag are used
                 to update the changelog in the spec-file.
         """
-        comment = self.entry_from_action or (
-            self.up.local_project.git_project.get_release(
-                tag_name=upstream_tag, name=full_version
-            ).body
-            if self.package_config.copy_upstream_release_description
-            else self.up.get_commit_messages(
-                after=self.up.get_last_tag(upstream_tag), before=upstream_tag
-            )
-        )
         try:
-            self.dg.set_specfile_content(self.up.specfile, full_version, comment)
+            if self.dg.specfile.has_autochangelog():
+                comment = None
+            elif self.entry_from_action:
+                comment = self.entry_from_action
+            elif self.package_config.copy_upstream_release_description:
+                comment = self.up.local_project.git_project.get_release(
+                    tag_name=upstream_tag, name=full_version
+                ).body
```

Comment:
This is not identical from a logical point of view with the previous version, and I think it would also run the action twice (which might have some side effects). I suggest keeping the original version with `or`:
```suggestion
            elif self.package_config.copy_upstream_release_description:
                comment = self.entry_from_action or self.up.local_project.git_project.get_release(
                    tag_name=upstream_tag, name=full_version
                ).body
```

Reply:
Since it is not safe to call self.entry_from_action twice and since the code should remain like the old one;
I changed once more the code so we do not need to broader the try/except block.